src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -2,24 +2,181 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
 using System.Diagnostics;
-using System.Net.Internals;
 using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
 using System.Runtime.InteropServices;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace System.Net
 {
     internal static partial class NameResolutionPal
     {
-        public const bool SupportsGetAddrInfoAsync = false;
+        private static volatile bool s_initialized;
+        private static readonly object s_initializedLock = new object();
 
-        public static void EnsureSocketsAreInitialized() { } // No-op for Unix
+        private static readonly unsafe Interop.Sys.GetHostEntryForNameCallback s_getHostEntryForNameCallback = GetHostEntryForNameCallback;
+        private static bool s_platformSupportsGetAddrInfoAsync;
 
-        internal static Task GetAddrInfoAsync(string hostName, bool justAddresses) =>
-            throw new NotSupportedException();
+        public static void EnsureSocketsAreInitialized()
+        {
+            // Actually a No-op for Unix as sockets don't need a startup, but used
+            // to check whether async address resolution is available or not.
+
+            if (!s_initialized)
+            {
+                Initialize();
+            }
+
+            static void Initialize()
+            {
+                lock (s_initializedLock)
+                {
+                    if (!s_initialized)
+                    {
+                        s_platformSupportsGetAddrInfoAsync = Interop.Sys.PlatformSupportsGetAddrInfoAsync();
+                        s_initialized = true;
+                    }
+                }
+            }
+        }
```

Comment:
Why not just do this in a static field initializer?
```C#
private static readonly bool s_supportsGetAddrInfoAsync = Interop.Sys.PlatformSupportsGetAddrInfoAsync();
```
?

Reply:
Just to have it the same as the windows-implementation. But I see (now) that there is a need to start the socket, which isn't needed on linux.

I'll update to this.